src/compiler/types.ts

```diff
@@ -2766,7 +2766,8 @@ namespace ts {
         getApparentType(type: Type): Type;
         getSuggestionForNonexistentProperty(node: Identifier, containingType: Type): string | undefined;
         getSuggestionForNonexistentSymbol(location: Node, name: string, meaning: SymbolFlags): string | undefined;
-        /* @internal */ getBaseConstraintOfType(type: Type): Type | undefined;
+        getBaseConstraintOfType(type: Type): Type | undefined;
+        getDefaultFromTypeParameter(type: Type): Type | undefined;

```

Comment:
Why not just take a `TypeParameter` as input?

Reply:
Just to be the same as `getBaseConstraintOfType`. Which, BTW, `getBaseConstraintOfType` technically returns the `resolvedBaseConstraint`, while `getConstraintFromTypeParameter` would actually return the `constraint`; the different being that `getBaseConstraintOfType` follows all the constraints on your constraints to get the actual final constraint, while `getConstraintFromTypeParameter` is just the immediate constraint. Should we be exposing both?